physical/mssql.go

```diff
@@ -0,0 +1,191 @@
+package physical
+
+import (
+	"database/sql"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/armon/go-metrics"
+	_ "github.com/denisenkom/go-mssqldb"
+	log "github.com/mgutz/logxi/v1"
+)
+
+type MsSQLBackend struct {
+	dbTable    string
+	client     *sql.DB
+	statements map[string]*sql.Stmt
+	logger     log.Logger
+}
+
+func newMsSQLBackend(conf map[string]string, logger log.Logger) (Backend, error) {
+	username, ok := conf["username"]
+	if !ok {
+		username = ""
+	}
+
+	password, ok := conf["password"]
+	if !ok {
+		password = ""
+	}
+
+	server, ok := conf["server"]
+	if !ok || server == "" {
+		return nil, fmt.Errorf("missing server")
+	}
+
+	database, ok := conf["database"]
+	if !ok {
+		database = "Vault"
+	}
+
+	table, ok := conf["table"]
+	if !ok {
+		table = "Vault"
+	}
+
+	appname, ok := conf["appname"]
+	if !ok {
+		appname = "Vault"
+	}
+
+	connectionTimeout, ok := conf["connectiontimeout"]
+	if !ok {
+		connectionTimeout = "30"
+	}
+
+	logLevel, ok := conf["loglevel"]
+	if !ok {
+		logLevel = "0"
+	}
+
+	connectionString := "server=" + server + ";app name" + appname + ";connection timeout=" + connectionTimeout + ";log=" + logLevel
+	if username != "" {
+		connectionString += ";user id=" + username
+	}
```

Comment:
After some further internal discussions, we believe this is fine.  Even though SSO makes a lot of sense in a user workstation context it could also work in this context.  This is especially true since the same person configuring the Vault installation will also be in charge of the physical backend configuration.